Compatibility with numpy 1.20.0

Just as a heads up, apart from a bunch of deprecation warnings when running the tests, there are also a few failures in the kartothek test suite when using the newly release numpy 1.20.0.

This is caused by https://github.com/numpy/numpy/issues/18287, which is already fixed and will probably be in an upcoming 1.20.1 release (but can also be fixed by changing `np.unicode` to `str`)

Example failure:

```
 ____________ ERROR collecting tests/io/dask/dataframe/test_read.py _____________
tests/io/dask/dataframe/test_read.py:185: in <module>
    @pytest.mark.parametrize("col", get_dataframe_not_nested().columns)
kartothek/core/testing.py:65: in get_dataframe_not_nested
    "unicode": pd.Series(["Ö"], dtype=np.unicode),
/opt/conda/envs/arrow/lib/python3.7/site-packages/pandas/core/series.py:335: in __init__
    data = sanitize_array(data, index, dtype, copy, raise_cast_failure=True)
/opt/conda/envs/arrow/lib/python3.7/site-packages/pandas/core/construction.py:480: in sanitize_array
    subarr = _try_cast(data, dtype, copy, raise_cast_failure)
/opt/conda/envs/arrow/lib/python3.7/site-packages/pandas/core/construction.py:587: in _try_cast
    maybe_cast_to_integer_array(arr, dtype)
/opt/conda/envs/arrow/lib/python3.7/site-packages/pandas/core/dtypes/cast.py:1723: in maybe_cast_to_integer_array
    casted = np.array(arr, dtype=dtype, copy=copy)
E   ValueError: invalid literal for int() with base 10: 'Ö'
```

Noticed this in the pyarrow nightly integration builds, see https://issues.apache.org/jira/browse/ARROW-11472. And since I did the effort to figure out the cause, I thought to spare you the time ;)

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -23,7 +23,7 @@ jobs:
       matrix:
         numfocus_nightly: [false]
         os: ["ubuntu-latest"]
-        pyarrow: ["0.17.1", "1.0.0", "2.0.0", "nightly"]
+        pyarrow: ["0.17.1", "1.0.1", "2.0.0", "nightly"]
         python: ["3.6", "3.7", "3.8"]
         include:
           - numfocus_nightly: true

--- a/conda-requirements.txt
+++ b/conda-requirements.txt
@@ -4,7 +4,7 @@ msgpack-python>=0.5.2
 # Currently dask and numpy==1.16.0 clash
 numpy!=1.15.0,!=1.16.0
 pandas>=0.23.0, !=1.0.0
-pyarrow>=0.17.1, <3
+pyarrow>=0.17.1,!=1.0.0, <3
 simplejson
 simplekv
 storefact

--- a/kartothek/core/testing.py
+++ b/kartothek/core/testing.py
@@ -62,7 +62,7 @@ def get_dataframe_not_nested():
             "float64": pd.Series([1.0], dtype=np.float64),
             "date": pd.Series([date(2018, 1, 1)], dtype=object),
             "datetime64": pd.Series(["2018-01-01"], dtype="datetime64[ns]"),
-            "unicode": pd.Series(["Ö"], dtype=np.unicode),
+            "unicode": pd.Series(["Ö"], dtype=str),
             "null": pd.Series([None], dtype=object),
             # Adding a byte type with value as byte sequence which can not be encoded as UTF8
             "byte": pd.Series([gen_uuid_object().bytes], dtype=object),

--- a/kartothek/serialization/testing.py
+++ b/kartothek/serialization/testing.py
@@ -55,7 +55,7 @@ def get_dataframe_not_nested(n):
                 [datetime(2018, 1, x % 31 + 1) for x in range(1, n + 1)],
                 dtype="datetime64[ns]",
             ),
-            "unicode": pd.Series([str(x) for x in range(n)], dtype=np.unicode),
+            "unicode": pd.Series([str(x) for x in range(n)], dtype=str),
             "null": pd.Series([None] * n, dtype=object),
             "bytes": pd.Series(binaries, dtype=np.object),
         }

--- a/requirements.txt
+++ b/requirements.txt
@@ -4,7 +4,7 @@ msgpack>=0.5.2
 # Currently dask and numpy==1.16.0 clash
 numpy!=1.15.0,!=1.16.0
 pandas>=0.23.0, !=1.0.0
-pyarrow>=0.17.1, <3
+pyarrow>=0.17.1,!=1.0.0, <3
 simplejson
 simplekv
 storefact

